sos/cleaner/__init__.py

```diff
@@ -218,59 +237,26 @@ def inspect_target_archive(self):
 
         In the event the target path is not an archive, abort.
         """
-        if not tarfile.is_tarfile(self.opts.target):
-            self.ui_log.error(
-                "Invalid target: must be directory or tar archive"
-            )
-            self._exit(1)
-
-        archive = tarfile.open(self.opts.target)
-        self.arc_name = self.opts.target.split('/')[-1].split('.')[:-2][0]
-
-        try:
-            archive.getmember(os.path.join(self.arc_name, 'sos_logs'))
-        except Exception:
-            # this is not an sos archive
-            self.ui_log.error("Invalid target: not an sos archive")
-            self._exit(1)
-
-        # see if there are archives within this archive
-        nested_archives = []
-        for _file in archive.getmembers():
-            if (re.match('sosreport-.*.tar', _file.name.split('/')[-1]) and not
-                    (_file.name.endswith(('.md5', '.sha256')))):
-                nested_archives.append(_file.name.split('/')[-1])
-
-        if nested_archives:
-            self.log_info("Found nested archive(s), extracting top level")
-            nested_path = self.extract_archive(archive)
-            for arc_file in os.listdir(nested_path):
-                if re.match('sosreport.*.tar.*', arc_file):
-                    if arc_file.endswith(('.md5', '.sha256')):
-                        continue
-                    self.report_paths.append(os.path.join(nested_path,
-                                                          arc_file))
-            # add the toplevel extracted archive
-            self.report_paths.append(nested_path)
-        else:
-            self.report_paths.append(self.opts.target)
-
-        archive.close()
-
-    def extract_archive(self, archive):
-        """Extract an archive into our tmpdir so that we may inspect it or
-        iterate through its contents for obfuscation
-
-        Positional arguments:
-
-            :param archive:     An open TarFile object for the archive
-
-        """
-        if not isinstance(archive, tarfile.TarFile):
-            archive = tarfile.open(archive)
-        path = os.path.join(self.tmpdir, 'cleaner')
-        archive.extractall(path)
-        return os.path.join(path, archive.name.split('/')[-1].split('.tar')[0])
+        force = False
+        if self.opts.archive_type != 'auto':
+            check_type = self.opts.archive_type.replace('-', '_')
+            self.archive_types = [
+                a for a in self.archive_types if a.type_name == check_type
+            ]
```

Comment:
(it took me a while to understand this, hence the below request)

Could you add a comment explaining why it is a *list*? (b'cos `check_type == report` can mean `type_name = SoSReportArchive` **or** `SoSReportDirectory` - curiously it works well even for the current `SoSDirectoryArchive` value :-o)

Reply:
Ack.

Reply:
Instead of a comment, I refactored the logic here to be a bit more robust and readable.